### PR TITLE
Doc for modifying a module

### LINKED ISSUE #52: Documenting your updates (from modifying analysis)
Write docs for users to know what they need to document like the analysis README, Docker file (link to docker section for detailed instructions), code comments.

**Relevant Journey Maps**
[Notify disagreement and propose new solution](https://www.figma.com/file/qjvNYWo8Tkv6HZ5yd6i9oE/OpenScPCA-Journey-Mapping?type=whiteboard&node-id=255-4328&t=ElcuB4UWylzhAGj6-11)

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #52 

This PR adds a docs page emphasizing where to update documentation when modifying a module. It's pretty short to start, so please let me know if/where I can add more content.

### CODE REVIEW

**Comment by sjspielman** [comment]
Spelling errors should go away once #381 is merged.

File: docs/contributing-to-analyses/analysis-modules/modifying-a-module.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Modifying an existing module
+
+When modifying or adding to an existing module, it's important to document all changes you make.
+
+!!! note
+    When you file an issue to track your work modifying an analysis module, you will use the [`Update an analysis` issue template](../../communications-tools/github-issues/issue-templates.md#update-an-analysis).
+
+Please make sure to make the following updates when you work with an existing module:
```
Comment (allyhawkins) [inline docs/contributing-to-analyses/analysis-modules/modifying-a-module.md:8]:
```suggestion

If you are modifying a module that already exists, then you will first need to file an issue using the [`Update an analysis` issue template](../../communications-tools/github-issues/issue-templates.md#update-an-analysis). 
Then you can create a new branch and commit any code changes. 

Be sure that any new code changes are accompanied by updates to documentation. 
You will also want to capture any new software dependencies. 

The following changes should be included in any PR's that modify existing modules: 
```
I felt like the first sentence was repetitive and only focused on documentation when that's not the only thing captured here. Also not sure this is the right place for the note? I think that's a first step before doing any of this so I moved it around.

**Review by allyhawkins — COMMENTED** [review]
I think the content is mostly here, I just added some ideas for reorganization.

File: docs/contributing-to-analyses/analysis-modules/modifying-a-module.md (hunk 1 above)
Comment (sjspielman) [reply docs/contributing-to-analyses/analysis-modules/modifying-a-module.md:8]: > Also not sure this is the right place for the note? 

I too was not sure fwiw!

File: docs/contributing-to-analyses/analysis-modules/modifying-a-module.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Modifying an existing module
+
+
+If you are modifying a module that already exists, then you will first need to file an issue using the [`Update an analysis` issue template](../../communications-tools/github-issues/issue-templates.md#update-an-analysis).
+Then you can create a new branch and commit any code changes.
```
Comment (allyhawkins) [inline docs/contributing-to-analyses/analysis-modules/modifying-a-module.md:5]:
```suggestion
Then you can create a new branch and commit any code changes before filing a PR. 
```

File: docs/contributing-to-analyses/analysis-modules/modifying-a-module.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+This includes all of:
+    - Changes to the behavior of any existing script or notebooks you modify
+    - New scripts or notebooks you add to the module
+    - Changes to which input data the module consumes
+    - Changes to any output files (e.g., results or plots) the module generates
+    - Changes to the module's [compute requirements](../determining-requirements/determining-compute-requirements.md)
+    - Changes to instructions for setting up the module's run environment and/or running the module
+- Make sure any new dependencies have been added to the module's [software environment](../determining-requirements/determining-software-requirements.md).
+- Either update or add `README.md` files to any new sub-folders you add to the module.
+    - For example, if you added a new script to the `scripts` directory, be sure to add a `scripts/README.md` that describes the usage of the new script.
```
Comment (allyhawkins) [inline docs/contributing-to-analyses/analysis-modules/modifying-a-module.md:20]: They should probably also update any existing results on S3 if they are changed?

**Review by allyhawkins — COMMENTED** [review]
I think the only thing missing is making sure results are up to date.

File: docs/contributing-to-analyses/analysis-modules/modifying-a-module.md (hunk 3 above)
Comment (sjspielman) [reply docs/contributing-to-analyses/analysis-modules/modifying-a-module.md:20]: good call!

**Comment by sjspielman** [comment]
I added something about S3 towards the top, and kept the bullet list focused on docs. Let me know about wording!

**Review by allyhawkins — APPROVED** [review]
LGTM

**Pull request merged**